pkg/controller/elasticsearch/settings/merged_config.go

```diff
@@ -111,7 +111,7 @@ func xpackConfig(ver version.Version, httpCfg commonv1.HTTPConfig) *CanonicalCon
 		cfg[esv1.XPackSecurityAuthcRealmsNativeNative1Order] = -99
 	}
 
-	if ver.IsSameOrAfter(version.MustParse("7.6.0")) {
+	if ver.IsSameOrAfter(version.MustParse("7.8.1")) {
 		cfg[esv1.XPackLicenseUploadTypes] = []string{
```

Comment:
I'm wondering what will happen if people do a version upgrade from 7.7.0 (platinum license) to 7.8.1 (requires an Enterprise license). Can they just continue with the existing platinum license, and only a call to refresh the license would be impacted?

Reply:
This could potentially lead to unpleasant errors in the logs. If the license was still valid everything would be fine. But once it expires or gets close to expire the operator would still try to apply a platinum license to the 7.8.1 cluster which would fail because of the restriction.  

I think we should at least not try to apply a platinum license.